iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/TopologyModificationUtils.java

```diff
@@ -494,47 +543,41 @@ private static List<Integer> getOrderPositions(ConnectablePosition<?> position,
-        List<Integer> orders = new ArrayList<>();
+    private static List<ConnectablePosition.Feeder> getBranchFeeders(ConnectablePosition<?> position, VoltageLevel voltageLevel, Branch<?> branch) {
+        List<ConnectablePosition.Feeder> feeders = new ArrayList<>();
         if (branch.getTerminal1().getVoltageLevel() == voltageLevel) {
-            Optional.ofNullable(position.getFeeder1()).flatMap(ConnectablePosition.Feeder::getOrder).ifPresent(orders::add);
+            Optional.ofNullable(position.getFeeder1()).ifPresent(feeders::add);
         }
         if (branch.getTerminal2().getVoltageLevel() == voltageLevel) {
-            Optional.ofNullable(position.getFeeder2()).flatMap(ConnectablePosition.Feeder::getOrder).ifPresent(orders::add);
+            Optional.ofNullable(position.getFeeder2()).ifPresent(feeders::add);
         }
-        checkConnectableInVoltageLevel(orders, voltageLevel, branch, throwException, reporter);
-        Collections.sort(orders);
-        return orders;
+        return feeders;
     }
 
-    private static List<Integer> get3wtOrders(ConnectablePosition<?> position, VoltageLevel voltageLevel, ThreeWindingsTransformer twt, boolean throwException, Reporter reporter) {
-        List<Integer> orders = new ArrayList<>();
+    private static List<ConnectablePosition.Feeder> get3wtFeeders(ConnectablePosition<?> position, VoltageLevel voltageLevel, ThreeWindingsTransformer twt) {
+        List<ConnectablePosition.Feeder> feeders = new ArrayList<>();
         if (twt.getLeg1().getTerminal().getVoltageLevel() == voltageLevel) {
-            Optional.ofNullable(position.getFeeder1()).flatMap(ConnectablePosition.Feeder::getOrder).ifPresent(orders::add);
+            Optional.ofNullable(position.getFeeder1()).ifPresent(feeders::add);
         }
         if (twt.getLeg2().getTerminal().getVoltageLevel() == voltageLevel) {
-            Optional.ofNullable(position.getFeeder2()).flatMap(ConnectablePosition.Feeder::getOrder).ifPresent(orders::add);
+            Optional.ofNullable(position.getFeeder2()).ifPresent(feeders::add);
         }
         if (twt.getLeg3().getTerminal().getVoltageLevel() == voltageLevel) {
-            Optional.ofNullable(position.getFeeder3()).flatMap(ConnectablePosition.Feeder::getOrder).ifPresent(orders::add);
+            Optional.ofNullable(position.getFeeder3()).ifPresent(feeders::add);
         }
-        checkConnectableInVoltageLevel(orders, voltageLevel, twt, throwException, reporter);
-        Collections.sort(orders);
-        return orders;
+        return feeders;
     }
 
-    private static void checkConnectableInVoltageLevel(List<Integer> orders, VoltageLevel voltageLevel, Connectable<?> connectable, boolean throwException, Reporter reporter) {
+    private static void checkOrdersExist(List<Integer> orders, Connectable<?> connectable, boolean throwException, Reporter reporter) {
```

Comment:
This is called when collecting all the orders of a slice or of a voltage level, I don't think this check is needed. We could rather do a higher level check on whether there's no orders at all on the slice / voltage level